tests/Avalonia.Controls.UnitTests/Presenters/ContentPresenterTests_Layout.cs

```diff
@@ -233,7 +233,7 @@ public void Child_Arrange_With_Zero_Height_When_Padding_Height_Greater_Than_Chil
 
             target.Arrange(new Rect(0, 0, 100, 100));
 
-            Assert.Equal(new Rect(48, 48, 0, 0), content.Bounds);
+            Assert.Equal(new Rect(32, 32, 0, 0), content.Bounds);
```

Comment:
This was correct before.

The content is h/w 0, but its Horizontal and Vertical content alignment = center, its placed inside a 100,100 box.
position should be ~50, so previous value of 48,48 was correct.

If this test isnt passing now, its probably ignoring horizontal and vertical content alignment.

Reply:
@Gillibald a tip: when not sure about this, create a new WPF/UWP project and add the controls and see how they're arranged there.

Reply:
The ContentPresenter never gets the space of 100x100 the initial value is 32x32 because is has a MaxHeight/MaxWidth of 32. So a child of zero size never can increase that. I can't see why it should end up at 48. The test for Border has the right values too. A padding of 32 in a box of 32 always leaves zero space. But let me compare it to the unchanged code first.

Reply:
Yeah what your saying makes sense, so I guess there must be a hug in existing implementation causing that to pass. I'll debug the test tomorrow to confirm.

Reply:
I have compared avalonia to uwp and they both produce kinda the same but avalonia ignores the case when there is no space left to place a control in. It always considers a controls set width and height thats kinda strange but thats another problem. So if there is zero space and a control's width/height is set to 5 the control is still visible.

Reply:
@danwalmsley did you get anywhere in your investigation?